### PR TITLE
🧪 Improve test coverage for collections JSON validation

### PR DESCRIPTION
**Pull request by is0692vs**
🎯 **What:** Improved test coverage for handling invalid JSON payloads across collection management endpoints.
  
📊 **Coverage:** Added explicit test scenarios to cover malformed JSON strings, non-object arrays, and missing/empty array arrays for:
- `POST /api/collections`
- `PATCH /api/collections/:id`
- `POST /api/collections/:id/papers`
- `PATCH /api/collections/:id/papers` (bulk edit)
  
✨ **Result:** Test coverage for validation lines has been brought to 100%, and previously untested `catch` blocks and edge conditions correctly reject invalid bodies as expected.

---
*PR created automatically by Jules for task [12105179442170494420](https://jules.google.com/task/12105179442170494420) started by @is0692vs*

<details open><summary><h3>Greptile Summary</h3></summary>

コレクション管理エンドポイント4つ（`POST /api/collections`、`PATCH /api/collections/:id`、`POST /api/collections/:id/papers`、`PATCH /api/collections/:id/papers`）に対するJSONバリデーションのテストカバレッジを強化するPRです。

- 各エンドポイントに「不正JSON文字列」「非オブジェクト配列」のテストケースを追加し、catchブロックおよびisArray判定の両方をカバー。
- `PATCH /api/collections/:id/papers` については `paper_ids` 欠落・空配列それぞれを独立した `it` ブロックに分割して追加（前回レビュー指摘の対応済み）。
- 既存テストの `body: \"invalid-json\"` を `body: \"invalid json {\"` に変更し、`Content-Type: application/json` ヘッダーの不足を補完。
</details>

<details open><summary><h3>Confidence Score: 5/5</h3></summary>

テストファイルのみの変更であり、プロダクションコードへの影響はありません。

変更はテストコードのみ。追加されたテストケースはすべて実装の実際の挙動と整合しており、モック順序の軽微な不一致を除いて論理的な問題は見当たりません。

特にありません。
</details>

<details open><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| apps/api/src/routes/__tests__/collections.test.ts | コレクションエンドポイント4つに対するJSONバリデーションのテストカバレッジを拡充。不正JSON・非オブジェクト配列・paper_ids欠落/空配列の各シナリオを追加。論理的な欠陥はなく、一部テストで queueSelectResponses の呼び出し順序がファイル内パターンと不一致。 |

</details>

</details>

<details><summary>Prompt To Fix All With AI</summary>

`````markdown
Fix the following 2 code review issues. Work through them one at a time, proposing concise fixes.

---

### Issue 1 of 2
apps/api/src/routes/__tests__/collections.test.ts:1893-1908
`queueSelectResponses` の呼び出し順序が既存のパターンと一致していません。このテストファイルでは `queueSelectResponses` を `createTestApp()` / `createTestEnv()` より**前**に呼ぶのが確立済みのパターンですが、この2つのテストでは逆になっています。現時点では `drizzle()` が `mockDb` への参照を返すため機能的には問題ありませんが、将来的にセットアップ順序に依存する変更が入った場合にサイレントに壊れる可能性があります。

```suggestion
  it("PATCH /api/collections/:id/papers rejects when paper_ids is missing", async () => {
    const token = await createTestJWT({ sub: "user-1" });

    queueSelectResponses([
      {
        getResult: {
          id: "col-1",
          ownerType: "user",
          ownerId: "user-1",
          visibility: "private",
        },
      },
    ]);

    const app = await createTestApp();
    const env = createTestEnv();
```

### Issue 2 of 2
apps/api/src/routes/__tests__/collections.test.ts:1927-1942
同様に `paper_ids` が空配列のテストでも `queueSelectResponses` が `createTestApp()` / `createTestEnv()` の後で呼ばれています。ファイル全体のパターン（`queueSelectResponses` → `createTestApp()` → `createTestEnv()`）に合わせることを推奨します。

```suggestion
  it("PATCH /api/collections/:id/papers rejects when paper_ids is empty", async () => {
    const token = await createTestJWT({ sub: "user-1" });

    queueSelectResponses([
      {
        getResult: {
          id: "col-1",
          ownerType: "user",
          ownerId: "user-1",
          visibility: "private",
        },
      },
    ]);

    const app = await createTestApp();
    const env = createTestEnv();
```

`````

</details>

<sub>Reviews (2): Last reviewed commit: ["Address collections JSON validation test..."](https://github.com/hiroki-org/openshelf/commit/294ad922389194184bf979cde97c6922265b9ebc) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=32547362)</sub>

### CODE REVIEW

File: apps/api/src/routes/__tests__/collections.test.ts [hunk 1]
```diff
@@ -402,6 +402,28 @@ describe("collections routes", () => {
...
+          Authorization: `Bearer ${token}`,
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify([]),
+      },
+      env as any,
+    );
+
+    expect(res.status).toBe(400);
+    expect(((await res.json()) as any).error).toBe("Invalid JSON body");
```
[the remark this answers is not in the document]
Comment (is0692vs) [reply apps/api/src/routes/__tests__/collections.test.ts]: 対応しました: JSON body assertionを await expect(res.json()).resolves.toEqual({ error: "Invalid JSON body" }) に変更しました。検証: npm --workspace apps/api run test -- src/routes/__tests__/collections.test.ts, npm run lint, npm run typecheck.

File: apps/api/src/routes/__tests__/collections.test.ts [hunk 2]
```diff
@@ -1770,6 +1881,69 @@ describe("collections routes", () => {
...
+          Authorization: `Bearer ${token}`,
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({ paper_ids: [] }),
+      },
+      env as any,
+    );
+
+    expect(res.status).toBe(400);
+    await expect(res.json()).resolves.toEqual({ error: "paper_ids is required" });
```
[the remark this answers is not in the document]
Comment (is0692vs) [reply apps/api/src/routes/__tests__/collections.test.ts]: 対応しました: paper_ids欠落と空配列を別々のitブロックに分割しました。検証: npm --workspace apps/api run test -- src/routes/__tests__/collections.test.ts, npm run lint, npm run typecheck.

**Comment by is0692vs** [comment]
@greptile review
